xapian-applications/omega/query.cc

```diff
@@ -1789,7 +1789,15 @@ eval(const string &fmt, const vector<string> &param)
 		    value = eval(args[2], param);
 		break;
 	    case CMD_include:
-		value = eval_file(args[0]);
+		try {
+		    value = eval_file(args[0]);
+		} catch (...) {
+		    if (args.size() == 2) {
+			value = eval(args[1], param);
+		    } else {
+			throw;
+		    }
+		}
```

Comment:
I don't think this should work via exception handling.  One issue is that this treats any exception the same, whereas I think really the case we're interested in here is when the file isn't present (or maybe is there but not readable).  Also it would be reasonable to use this to include an optional template (kind of like a hook point), and so that optional template might never exist for a particular site, so the "file not found" case really isn't exceptional.

C++ exceptions are implemented on the assumption that they rarely happen and the design puts the cost when an exception is thrown/handled, and not for the no-exception case.